src/elevator/ElevatorSystem.java

```diff
@@ -1,58 +1,77 @@
 package elevator;
 
-import java.util.ArrayList;
 import java.util.LinkedList;
-import java.util.List;
 import java.util.Queue;
 
 import common.Message;
-import common.Requestable;
+import common.MessageType;
 import scheduler.Scheduler;
 
+/**
+ * System for managing elevators and receive messages from scheduler
+ * 
+ * @author Derek Shao, Souheil Yazji
+ *
+ */
 public class ElevatorSystem implements Runnable {
-	
-	private Requestable scheduler; 
-	private List<Elevator> elevators;
-	private Queue<Message> requests;
-	
+
+	private Scheduler scheduler;
+	private Elevator ele1;
+	private Queue<Message> inBoundRequests, outBoundRequests;
+
 	public ElevatorSystem(Scheduler scheduler) {
 		this.scheduler = scheduler;
-		this.elevators = new ArrayList<Elevator>();
-		this.requests = new LinkedList<Message>();
-		this.addElevator();
+		this.inBoundRequests = new LinkedList<Message>();
+		this.outBoundRequests = new LinkedList<Message>();
+		this.ele1 = new Elevator(10, 0, this);
 		startSystem();
 	}
-	
-	public void addElevator() {
-		this.elevators.add(new Elevator(10, 0, this, scheduler));
-		this.elevators.add(new Elevator(10, 1, this, scheduler));
-	}
-	
+
+	/**
+	 * Start the elevators
+	 */
 	public void startSystem() {
-		for (Elevator ele : this.elevators) {
-			Thread eleThread = new Thread(ele);
-			eleThread.start();
-		}
+		Thread eleThread = new Thread(this.ele1, "Elevator");
+		eleThread.start();
 	}
-	
-	public synchronized void addMessage(Message msg) {
-		synchronized(requests) {
-			this.requests.add(msg);
-		}
-	} 
 
 	@Override
 	public void run() {
-		
-		synchronized(requests) {
-			while (this.requests.isEmpty()) {
-				try {
-					wait();
-				} catch (InterruptedException e) {
-					// TODO Auto-generated catch block
-					e.printStackTrace();
+		while (true) {
+			while (!inBoundRequests.isEmpty() || !outBoundRequests.isEmpty()) {
+				while (!inBoundRequests.isEmpty()) {
+					System.out.println("Elevator System: Sending messages to free elevator");
+					ele1.request(inBoundRequests.poll());
 				}
+				
+				synchronized (outBoundRequests) {
```

Comment:
So you can't send messages to both elevators and scheduler at the same time. For now I think it's fine but I think  ideally we should have two threads.